modules/shared/facts.d/win_env_fact.ps1

```diff
@@ -0,0 +1,27 @@
+# Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Creates useful facts based off of system environment variables
+
+$systemdrive     = $env:systemdrive
+$system32        = "$env:systemdrive\\windows\\system32"
```

Comment:
Instead of executing $env again just to get the system drive, why not use the puppet var to define $system32.  I would also add the suffix _path to any path you are defining so this would be $system32_path.

Reply:
Done.